PowerEditor/src/tools/NppShell/Installer.cpp

```diff
@@ -0,0 +1,326 @@
+#include "pch.h"
+#include "Installer.h"
+
+#include "EditWithNppExplorerCommandHandler.h"
+
+#define GUID_STRING_SIZE 40
+
+using namespace winrt::Windows::ApplicationModel;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+
+using namespace NppShell::Helpers;
+using namespace NppShell::Installer;
+
+const wstring SparsePackageName = L"NotepadPlusPlus";
+const int FirstWindows11BuildNumber = 22000;
+
+#ifdef WIN64
+const wstring ShellExtensionKey = L"Software\\Classes\\*\\shell\\ANotepad++64";
+#else
+const wstring ShellExtensionKey = L"Software\\Classes\\*\\shell\\ANotepad++";
+#endif
+
+struct DOREGSTRUCT {
+    HKEY hive;
+    wstring key;
+    wstring name;
+    wstring value;
+};
+
+bool IsWindows11Installation()
```

Comment:
The proper way to check for this is calling VerifyVersionInfo: https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-verifyversioninfow

Reply:
I did look at that first, but it seems like that function is deprecated from Windows 10 going forward.

> Note: This function has been deprecated for Windows 10. See [targeting your applications for Windows](https://learn.microsoft.com/en-us/windows/win32/sysinfo/targeting-your-application-at-windows-8-1) for more information.

That is why I read the build number from the registry, since that is supported.

Reply:
Reading from the registry isn't supported, no. There is no docs telling you that this registry key is a supported way for apps to get the OS build number.

Also the function is deprecated, sure, but they point you to version helper macros as the non-deprecated replacement, which ironically makes use of VerifyVersionInfo. So it isn't truly deprecated 😂